File: leetcode-submissions/3016_Minimum_Number_of_Pushes_to_Type_Word_II.ts
function minimumPushes(word: string): number {
    let freq:number[]=Array(26).fill(0);
    for(const ch of word){
         const index = ch.charCodeAt(0) - 'a'.charCodeAt(0);
        freq[index]=freq[index]===0?1:freq[index]+1;
    }
    freq.sort((a,b)=>b-a);
    let ans:number=0;

    for(let i=0;i<26;i++){
        if(i<8){
            ans+=freq[i];
       }else if(i<16){
        ans+=freq[i]*2;

       }else if(i<24){
        ans+=freq[i]*3;
       }else{
        ans+=freq[i]*4;
       }
    }
    return ans;
};

File: leetcode-submissions/3016_Minimum_Number_of_Pushes_to_Type_Word_II.cpp
class Solution {
public:
    int minimumPushes(string word) {
        vector<int>freq(26,0);
        for(char ch:word){
            freq[ch-'a']++;
        }
        int ans =0;
sort(freq.begin(),freq.end(),greater<int>());//sorting in decreasing order because we need to use greedy approach 
        for(int i=0;i<26;i++){
            if(i<8){
                ans+=freq[i];
            }else if( i<16){
                ans+=freq[i]*2;
            }else if(i<24){
                ans+=freq[i]*3;
            }else {
                ans+=freq[i]*4;
            }
        }
        return ans ;
    }
};